encounter.md

```diff
@@ -0,0 +1,258 @@
+    *visit* | `Visit UUID` | If allocating an encounter for an already existing visit should specify the target visit UUID 
+   
+    ```console
+        POST /encounter
+        {
+          "encounterDatetime": "2019-10-16 12:08:43",
+          "patient": "070f0120-0283-4858-885d-a20d967729cf",
+          "encounterType": "e22e39fd-7db2-45e7-80f1-60fa0d5a4378",
+          "location": "aff27d58-a15c-49a6-9beb-d30dcfc0c66e",
+          "encounterProviders": [
+            {
+              "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
+              "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
+            }
+          ]
+        }
+    ```
+### Update an encounter
+
+*  Update a target encounter with given UUID, this method only modifies properties in the request. Returns a `404 Not Found` 
+status if encounter not exists. If user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *uuid* | `target_encounter_uuid` | Target encounter resource UUID
+    
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *encounterDatetime* | `Date (ISO8601 Long)` | The date and time the encounter was created (required)
+    *patient* | `Patient UUID` | Patient type resource UUID (required)
+    *encounterType* | `EncounterType UUID` | Encounter type resource UUID (required)
+    *location* | `Location UUID` | Location resource UUID (required)
+    *encounterProviders* | `Array[]: Providers` | Array of Providers already registered in OpenMRS    
+    *obs* | `Array[]: Obs` | Array of observations and values for the encounter.    
+    *orders* | `Array[]: Order UUID` | List of orders created during the encounter    
+    *form* | `FormType UUID` | Form type to be filled for the encounter
+    *visit* | `Visit UUID` | If allocating an encounter for an already existing visit should specify the target visit UUID 
+    
+    ```console
+        POST /encounter/:target_encounter_uuid
+        {
+          "encounterDatetime": "2019-10-16 12:08:43",
+          "patient": "070f0120-0283-4858-885d-a20d967729cf",
+          "encounterType": "e22e39fd-7db2-45e7-80f1-60fa0d5a4378",
+          "location": "aff27d58-a15c-49a6-9beb-d30dcfc0c66e",
+          "encounterProviders": [
+            {
+              "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
+              "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
+            }
+          ]
+        }
+    ```
+    
+### Delete an encounter
+
+* Delete or Retire a target encounter by its UUID. Returns a `404 Not Found` status if encounter not exists.If user not logged 
+  in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *purge* | `Boolean` | The resource will be voided/retired unless purge = ‘true’
```

Comment:
Encounters are voided, not retired. Metadata (like concepts & encounter types) are retired when they should no longer be used. Data (like patients, encounter, observations, etc.) are voided when the user wants to "delete" (invalidate) them.